### PR TITLE
Jenkinsfile: use NFS-setup for yocto-cache

### PR DESCRIPTION
**Pull request by tobsan**
Instead of copying to /vagrant/archive and then depend on another job to
publish the cache, we assume /var/yocto-cache is a NFS share to which we
can write (and read from). The share should be shared with any build
slave we have.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -69,8 +69,8 @@ def buildManifest = {String bitbake_image, String bsp, boolean qtauto ->
 // Run the different jobs in parallel, on different slaves
 parallel 'intel':{
     node("DockerCI") { buildManifest("core-image-pelux-minimal", "intel", false) }
-},'intel-qtauto':{
+}/*,'intel-qtauto':{
```
Comment (rpannek) [inline Jenkinsfile]: This should go into a different commit.

File: site.conf.in [hunk 2]
```diff
@@ -1,3 +1,3 @@
 INHERIT += "own-mirrors"
-SOURCE_MIRROR_URL ?= "%CACHEURL%/downloads/"
-SSTATE_MIRRORS ?= "file://.* %CACHEURL%/sstate-cache/PATH"
+SOURCE_MIRROR_URL ?= "file://%CACHEURL%/downloads/"
```
Comment (rpannek) [inline site.conf.in]: Could you move it into the env.YOCTO_CACHE_URL? So this could then use both a NFS and HTTP cache.

File: Jenkinsfile (hunk 1 above)
Comment (tobsan) [reply Jenkinsfile]: It will be removed completely later on, so that it builds all images.

File: site.conf.in (hunk 2 above)
Comment (tobsan) [reply site.conf.in]: I suppose I could, but it would become problematic when trying to copy artifacts to the CACHE_URL (shells doesn't handle "file://")

Comment (tobsan) [reply site.conf.in]: But I suppose we could do a check to see if the CACHEURL begins with http:// or file:// and act accordingly, if we want to easily switch between the two. On the other hand, publishing to the http:// mirror has to be handled separately in that case anyway, which it isn't in the NFS case.
